### PR TITLE
Add collision detection and a simple gameover page

### LINKED ISSUE #6: Detect collision between bird and pipe

### PR DESCRIPTION
**Pull request by MubaiHua**
Need modification later, use current version to check feature

### CODE REVIEW

File: flappy-bird.js [hunk 1]
```diff
@@ -229,12 +263,47 @@ export class Bird extends Scene {
...
-        if(this.game_start && (t - this.elapsed_time_before_game_start) * this.game_speed / this.pipe_distance > 5){
+        if (this.game_start && (t - this.elapsed_time_before_game_start) * this.game_speed / this.pipe_distance > 5) {
             const starting_pipe_model_transform3 = model_transform.times(Mat4.translation(0, 10, pipe_pos3));
             this.draw_all_pipe(context, program_state, starting_pipe_model_transform3);
         }
     }
 
+
+    isCollision(pipe_position) {
+        const rx = pipe_position.rx, ry = pipe_position.ry, rw = pipe_position.rw, rh = pipe_position.rh;
```
Comment (lorenzoc25) [inline flappy-bird.js]: `const { rx, ry, rw, rh } = pipe_position`

File: flappy-bird.js [hunk 2]
```diff
@@ -229,12 +263,47 @@ export class Bird extends Scene {
...
+        // get distance from the closest edges
+        const distX = cx - testX;
+        const distY = cy - testY;
+        const distance = Math.sqrt( (distX * distX) + (distY * distY) );
+        //const myObj = { pipey : testY,pipex : testX, birdy : cy };
+        //console.log(myObj);
+
+        // if the distance is less than the radius, collision!
+        return distance <= radius;
+    }
```
Comment (lorenzoc25) [inline flappy-bird.js:307]: variable names are still not clear enough. Better change it up

File: flappy-bird.js [hunk 3]
```diff
@@ -198,18 +207,43 @@ export class Bird extends Scene {
...
             this.draw_pipe(context, program_state, bottom_pipe_model_transform, pipe_len);
 
-            //draw bottom pipe
+            //draw top pipe
             const top_pipe_model_transform = model_transform.times(Mat4.translation(0, this.pipe_gap - (9 - pipe_len), i * this.pipe_distance))
-                .times(Mat4.rotation(Math.PI, 1, 0, 0));
+                                                            .times(Mat4.rotation(Math.PI, 1, 0, 0));
             this.draw_pipe(context, program_state, top_pipe_model_transform, 9 - pipe_len);
+
+            //determine collision on top and bottom
```
Comment (lorenzoc25) [inline flappy-bird.js]: The whole logic can be abstract to a function, say `checkCollision(top_pipe, bottom_pipe)`

**Review by lorenzoc25 — CHANGES_REQUESTED** [review]

File: flappy-bird.js (hunk 1 above)
Comment (MubaiHua) [reply flappy-bird.js]: pro javascripter

**Pull request merged**